lib/modelManagers/followModel.js

```diff
@@ -0,0 +1,115 @@
+const { follows, User } = require('../../database/models');
+/**
+ * @description Follow and Unfollow user
+ *
+ * @class FollowModelManager
+ */
+class FollowModelManager {
+  /**
+   * @description Get a single User
+   * @static
+   * @param {any} id
+   * @returns {Object} - return user
+   *
+   * @memberOf FollowModelManager
+   */
+  static async getUserById(id) {
+    const getUser = await User.findOne({
+      where: {
+        id
+      }
+    });
+
+    return getUser;
+  }
+
+  /**
+   * @description handle follow user
+   * @static
+   * @param {any} followeeId
+   * @param {any} followerId
+   * @returns {object} return a new followee
+   *
+   * @memberOf FollowModelManager
+   */
+  static async followUser(followeeId, followerId) {
+    const follow = await follows.findOrCreate({
+      where: {
+        followeeId,
+        followerId
+      },
+      attributes: ['id', 'followerId', 'followeeId']
+    })
+      .spread((check, created) => created);
+    return follow;
+  }
+
+  /**
+ * @description Unfollow user
+ *
+ * @static
+ * @param {any} followerId
+ * @param {any} followeeId
+ * @returns {null} null
+ * @memberOf FollowModelManager
+ */
+  static async unfollowUser(followerId, followeeId) {
+    const unFollow = await follows.destroy({
+      where: {
+        followerId,
+        followeeId
+      }
+    });
+    return unFollow;
+  }
+
+  /**
+   * @description handle find and count numbers of occurances
+   *
+   * @static
+   * @param {any} followerId
+   * @returns {object} return followee and count all
+   * @memberOf FollowModelManager
+   */
+  static async getUserFollowee(followerId) {
+    const followees = await follows.findAndCountAll({
+      where: {
+        followerId
+      },
+      include: [
+        {
+          model: User,
+          as: 'myfolloweeId',
+          attributes: ['id', 'email', 'firstname', 'lastname']
+        }
+      ]
+    });
+    return followees;
+  }
+
+  /**
+   * @description handle find and count numbers of occurances
+   *
+   * @static
+   * @param {any} followeeId
+   * @returns {object} return follower and count all
+   * @memberOf FollowModelManager
+   */
+  static async getUserFollowing(followeeId) {
```

Comment:
`following` should be `follower`. Follower best describe the operation
https://www.google.com/search?q=follower&oq=follower&aqs=chrome..69i57j0l5.5297j0j7&sourceid=chrome&ie=UTF-8#dobs=follower